### PR TITLE
Added mTLS support

### PR DESCRIPTION
**Pull request by irineu**
When pass the CA file at last parameter on enableSSL method on server, the server will only accept the connection if the client send the CA certificate (and vice-versa). The server will check if the client public certificate is issued by the client CA certificate passed on enabledSSL param at the server, if not the server will drop the connection (and vice-versa).

This is a zero trust solution for network communication.

### CODE REVIEW

**Comment by irineu** [comment]
Idk what is the lint error on ubuntu gcc. the log does not provide many details.

**Comment by an-tao** [comment]
> Idk what is the lint error on ubuntu gcc. the log does not provide many details.

It's a check of code styles. you should install a clang-format(v11.0.1) on your system and run the format.sh before committing.
Or, I can do this for you.

**Review by marty1885 — COMMENTED** [review]
Sorry for the late reply, I'm more busy lately. Code looks almost ready. I've left a few commands for minor changes before we can merge. 

Thanks for the great PR!

File: CMakeLists.txt [hunk 1]
```diff
@@ -128,7 +128,7 @@ endif(WIN32)
 find_package(OpenSSL)
 if(OpenSSL_FOUND)
   target_link_libraries(${PROJECT_NAME} PRIVATE OpenSSL::SSL OpenSSL::Crypto)
-  target_compile_definitions(${PROJECT_NAME} PRIVATE USE_OPENSSL)
+  target_compile_definitions(${PROJECT_NAME} PRIVATE USE_OPENSSL) #ALLOW_CRT_LOCAL
```
[the remark this answers is not in the document]
Comment (irineu) [reply CMakeLists.txt:131]: I'll update the file to describe the use case of this flag.

File: trantor/net/inner/TcpConnectionImpl.cc [hunk 2]
```diff
@@ -356,14 +360,39 @@ std::shared_ptr<SSLContext> newSSLServerContext(
...
+        if (checkCA)
+        {
+            STACK_OF(X509_NAME) *cert_names =
+                SSL_load_client_CA_file(caPath.c_str());
+            if (cert_names != NULL)
+            {
+                SSL_CTX_set_client_CA_list(ctx->get(), cert_names);
+            }
+            ctx->mtlsEnabled = true;
+            // sk_X509_NAME_free(cert_names); //segmentation fault
```
[the remark this answers is not in the document]
Comment (irineu) [reply trantor/net/inner/TcpConnectionImpl.cc:378]: I've checked the example and the use case is when unsuccessfully call the function: SSL_add_dir_cert_subjects_to_stack for a certs directory. Add more than a CA certificate may be a nice feature in the future, but for while i'll just remove the commented code.

**Comment by marty1885** [comment]
@an-tao Is this PR ready to merge?

**Pull request merged**